src/lib.rs

```diff
@@ -569,6 +613,8 @@ pub struct PerCrateAnalysis {
     ref_spans: HashMap<Id, Vec<Span>>,
     globs: HashMap<Span, Glob>,
     impls: HashMap<Id, Vec<Span>>,
+    per_fn_borrows: HashMap<Id, BorrowData>,
+    fn_span_lookup_tree: BTreeMap<SearchSpan, Id>,
```

Comment:
> It's safer type-wise since one wouldn't be able to accidentally store the SpanKind's Start and End values like you can now

Is this invariant enforced dynamically in the current system?